### PR TITLE
Nav bar

### PR DESCRIPTION
**Pull request by Cililocwee**
## Type of change

[x] New feature
<br />
[] Bug fix
<br />
[] Documentation
<br />
[] Code refactor/tech debt
<br />
[] Infrastructure

## Description
This implements a responsive navbar that has a mobile view (drop down)

## Ticket link
[Ticket](https://linear.app/cdw/issue/CDW-43/style-the-navbar)

## Part of a back end implementation?
[x] No
<br />
[] [Yes]()  

## How to test

1. Checkout this branch and run frontend
2. Check nav buttons by navigating between pages
3. Resize window to check repsonsiveness
4. Toggle mobile view in devtools to see dropdown hamburger menu
5. Click hamburger menu to make sure dropdown toggles

## Screenshots

image
image
image

### CODE REVIEW

File: frontend/src/components/UserInterface.tsx [hunk 1]
```diff
@@ -3,14 +3,14 @@ import { Header } from "./header/Header";
 import { Outlet } from "react-router-dom";
 
 export const UserInterface = () => {
-	return (
-		<Container margin="auto" maxWidth="90%" padding={"28px 0px"}>
-			<Grid templateRows="repeat(3, auto)" gap={1}>
-				<Header />
+  return (
+    <Container margin="auto" maxWidth="90%" padding={"28px 0px"} pt={20}>
+      <Grid templateRows="repeat(3, auto)" gap={1}>
```
Comment (jtabba) [inline frontend/src/components/UserInterface.tsx:8]: Very good

File: frontend/src/components/header/Header.tsx [hunk 2]
```diff
@@ -2,23 +2,28 @@ import { Flex } from "@chakra-ui/react";
...
+      justifyContent="space-between"
+      alignItems="center"
+      h="60px"
+      outline={1}
+      position={"absolute"}
+      top={0}
+      left={0}
+      w={"100dvw"}
+      px={8}
+    >
```
Comment (jtabba) [inline frontend/src/components/header/Header.tsx:18]: Getting a bit chunky, lets export it into a local `styles.ts` folder

File: frontend/src/components/header/NavBar.tsx [hunk 3]
```diff
@@ -1,9 +1,152 @@
-import { Link } from "react-router-dom";
-
-export const NavBar = () => (
-	<>	
-		<Link to={"/explore"}>Explore</Link>
-		<Link to={"/products"}>Product</Link>
-		<Link to={"/mentors"}>Mentors</Link>
-	</>
-);
+"use client";
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: What's this for?

File: frontend/src/components/header/NavBar.tsx [hunk 4]
```diff
@@ -1,9 +1,152 @@
...
+        <Flex
+          bg={useColorModeValue("inherit", "gray.800")}
+          color={useColorModeValue("gray.600", "white")}
+          minH={"60px"}
+          py={{ base: 2 }}
+          px={{ base: 4 }}
+          borderBottom={1}
+          borderStyle={"solid"}
+          borderColor={useColorModeValue("gray.200", "gray.900")}
+          align={"center"}
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: Same with above

File: frontend/src/components/header/NavBar.tsx [hunk 5]
```diff
@@ -1,9 +1,152 @@
...
+  return (
+    <Stack bg={useColorModeValue("white", "gray.800")} display={{ md: "none" }}>
+      {NAV_ITEMS.map((navItem) => (
+        <MobileNavItem key={navItem.label} {...navItem} />
+      ))}
+    </Stack>
+  );
+};
+
+const MobileNavItem = ({ label, children, href }: NavItem) => {
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: Use FC

File: frontend/src/components/header/NavBar.tsx [hunk 6]
```diff
@@ -1,9 +1,152 @@
...
+  );
+};
+
+interface NavItem {
+  label: string;
+  subLabel?: string;
+  children?: Array<NavItem>;
+  href?: string;
+}
+
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: Move to top for easier readability with props

File: frontend/src/components/header/NavBar.tsx [hunk 7]
```diff
@@ -1,9 +1,152 @@
...
+      <Box
+        py={2}
+        as="a"
+        href={href ?? "#"}
+        justifyContent="space-between"
+        alignItems="center"
+        textAlign={"center"}
+        _hover={{
+          textDecoration: "none",
+        }}
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: Move to `styles` file

File: frontend/src/components/header/NavBar.tsx [hunk 8]
```diff
@@ -1,9 +1,152 @@
...
+            as="a"
+            p={2}
+            href={navItem.href ?? "#"}
+            fontSize={"sm"}
+            fontWeight={500}
+            color={linkColor}
+            _hover={{
+              textDecoration: "none",
+              color: linkHoverColor,
+            }}
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: `styles`

File: frontend/src/components/header/NavBar.tsx [hunk 9]
```diff
@@ -1,9 +1,152 @@
...
+  },
+  {
+    label: "Products",
+    href: "products",
+  },
+  {
+    label: "Mentors",
+    href: "mentors",
+  },
+];
```
Comment (jtabba) [inline frontend/src/components/header/NavBar.tsx]: I'd suggest moving to own file with the component that uses it

**Review by jtabba — CHANGES_REQUESTED** [review]
Great work overall Corrie, just a few things to tidy up.

Aside from that lets increase the size of the nav bar

**Comment by mrgiles1** [comment]
Really like the NAVBAR Corrie. My initial thoughts were:
1. I'm a fan of the standard, right-side hamburger menu on mobile
2. I generally like my links, currently centered on desktop, to be either up against the logo (left) or up against the profile icon (right)
3. I mentioned the hamburger button not being vertically aligned with the links (seems a touch lower than them).
Other than those ideas, suggestions, love it.

File: frontend/src/components/header/Header.tsx (hunk 2 above)
Comment (Cililocwee) [reply frontend/src/components/header/Header.tsx:21]: I'm trying to implement this change now, but a big problem I'm running into is that by styling the component with the style property overrides all the chakra stylings that I'm NOT styling:

When I use style.ts:
image

When I style the component using the chakra props:

image

**Comment by Cililocwee** [comment]
I've changed how the mobile menu works/looks/aligns:

image

The avatar/user profile icon is now tucked inside the menu:

image
image

The desktop menu is now bigger and easier to see:

image

**Review by jtabba — APPROVED** [review]

**Comment by Cililocwee** [comment]
Brought prod to branch, ready for clean merge

File: frontend/src/components/header/MobileNav.tsx [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  MenuButton,
+  Button,
+  MenuDivider,
+} from "@chakra-ui/react";
+import { FC } from "react";
+import { NAV_ITEMS, NavItem } from "./DesktopNav";
+import { AuthButtons } from "../authentication/SignInOrOut";
+import { CloseIcon, HamburgerIcon } from "@chakra-ui/icons";
+
+const MobileNavItem = ({ label, children, href }: NavItem) => {
```
Comment (jtabba) [inline frontend/src/components/header/MobileNav.tsx]: Keep it consistent and use FC

File: frontend/src/components/header/MobileNav.tsx [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+          fontSize={"lg"}
+        >
+          {label}
+        </Text>
+      </Box>
+    </Stack>
+  );
+};
+
+interface NavBarProps {}
```
Comment (jtabba) [inline frontend/src/components/header/MobileNav.tsx]: Whats this for?

**Review by jtabba — CHANGES_REQUESTED** [review]
Two small changes

File: frontend/src/components/header/MobileNav.tsx (hunk 11 above)
Comment (Cililocwee) [reply frontend/src/components/header/MobileNav.tsx]: Style and pizzazz!!.....

It was a fundamental misunderstanding of your request for me to use FC :) I'll prune it

File: frontend/src/components/header/MobileNav.tsx (hunk 10 above)
Comment (Cililocwee) [reply frontend/src/components/header/MobileNav.tsx]: Got it! Finally, actually got it!

**Comment by Cililocwee** [comment]
Ready for reconsideration

**Review by jtabba — APPROVED** [review]

**Pull request merged**